src/serac/physics/solid_functional.hpp

```diff
@@ -352,7 +521,14 @@ class SolidFunctional<order, dim, Parameters<parameter_space...>, std::integer_s
     residual_->AddBoundaryIntegral(
         Dimension<dim - 1>{},
         [this, traction_function](auto x, auto n, auto, auto, auto... params) {
-          return -1.0 * traction_function(x, n, time_, params...);
+
+          auto traction_vector = traction_function(x, n, time_, params...);
+
+          if (coordinate_transform_) {
+            traction_vector = dot(*coordinate_transform_, traction_vector);
+          }
```

Comment:
this pattern doesn't work on a GPU, using the `this` pointer to get at members of `SolidFunctional` inside the q-function requires dereferencing a host pointer (it's also won't work to pass `time_` either).